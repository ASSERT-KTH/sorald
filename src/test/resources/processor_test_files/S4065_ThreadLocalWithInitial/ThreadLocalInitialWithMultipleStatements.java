
public class ThreadLocalInitialWithMultipleStatements {
  
  public void bar() {
    ThreadLocal<String> threadLocal = new ThreadLocal<String>() { // Noncompliant
      @Override
      protected String initialValue() {
        String s = "hello";
        return s+"42";
      }
    };
    threadLocal.set("42");
  }
}
