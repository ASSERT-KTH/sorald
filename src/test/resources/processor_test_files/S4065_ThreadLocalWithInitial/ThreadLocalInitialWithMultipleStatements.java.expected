
public class ThreadLocalInitialWithMultipleStatements {
  
  public void bar() {
    ThreadLocal<String> threadLocal = ThreadLocal.withInitial(() -> {String s = "hello"; return s+"42";});
    threadLocal.set("42");
  }
}
